src/resolvers.ts

```diff
@@ -0,0 +1,147 @@
+import { Book, Customer, LendingRecord } from "./models";
+import { BooksDataSource } from "./data/booksDataSource";
+import { CustomersDataSource } from "./data/customersDataSource";
+import { logger } from "./logger";
+
+const booksDataSource = new BooksDataSource();
+const customersDataSource = new CustomersDataSource();
+
+export const resolvers = {
+  Query: {
+    // Get all books
+    books: (): Book[] => booksDataSource.getBooks(),
+
+    // Get a book by id
+    book: (_: any, { id }: { id: string }): Book | undefined => {
+      return booksDataSource.getBookById(id);
+    },
+
+    // Get all customers
+    customers: (): Customer[] => customersDataSource.getCustomers(),
+
+    // Get a customer by id
+    customer: (_: any, { id }: { id: string }): Customer | undefined =>
+      customersDataSource.getCustomerById(id),
+  },
+
+  Mutation: {
+    // Add a new book, the isLent field is set to false by default
+    addBook: (
+      _: any,
+      { id, title, author }: { id: string; title: string; author: string }
+    ): Book => {
+      const existingBook = booksDataSource.getBookById(id);
```

Comment:
Hmm I think that ID should not be provided when creating a new Book/Customer/Records

If we think a bit globally and pretend that data comes from the UI, if we are creating a new Book, the ID will not be known on the UI and should not be generated there.

For speed and reliability and security the BE(or the DB actually) should have a mechanism of generating unique IDs for every entry in every table.

Do you think we can have that here?

Reply:
awesome! that is exactly what we use here for similar cases 🔥